blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/util/FavoriteUtil.java

```diff
@@ -107,4 +128,20 @@ public static BlueFavorite getFavorite(Item item, @Nonnull Reachable parent){
         return null;
     }
 
+    /**
+     * Resolves the default branch for a folder
+     * @param folder to check within
+     * @return default branch
+     */
+    public static Job resolveDefaultBranch(AbstractFolder folder) {
+        // TODO: lookup the multibranch project and look for a default branch property
+        TopLevelItem job = folder.getJob(DEFAULT_BRANCH);
+        if(job == null) {
+            throw new ServiceException.BadRequestExpception("no master branch to favorite");
+        }
+        if (!(job instanceof Job)) {
+            throw new ServiceException.MethodNotAllowedException(DEFAULT_BRANCH + " is not a job");
```

Comment:
This will be a pretty hard failure from UX perspective - should it instead pick any branch, or perhaps fail silently?